src/semaphore/broadcast/markdown.py

```diff
@@ -142,6 +142,20 @@ def is_relevant_to_env(self, env_name: str) -> bool:
         else:
             return False
 
+    def summarize(
+        self, summary: Optional[str], body: Optional[str]
+    ) -> tuple[Optional[str], Optional[str]]:
```

Comment:
Thinking of the name, I think `extract_content` would be easier to understand, since it's dealing with both the summary and the body content.

Since this is a method, and `summary` will always be `self.metadata.summary` and `body` will always be `self.body`, it's preferable for this method to not have any arguments.

Lastly, the method _needs_ to always return a summary. That's why the returned summary has a type of `str` rather than `Optional[str]` (or `str | None` is the preferred way of typing a "nullable" return nowadays). What we can do is raise an exception if a summary cannot be extracted. There are better exceptions we could make, but a RuntimeError would do fine for now, as I'll show.

So then this becomes:

```suggestion
    def extract_content(self) -> tuple[str, Optional[str]]:
```

Lastly, I'll mention that I might take a slightly different approach, which is to have one method that gets the extracted summary, and one method that gets the extracted body. I get that this means more lines of code, but it avoids returning a tuple, which is something I seldom do because its a little more "vague". And since these are all methods that take no arguments, I'd probably even turn them into properties, like this:

```python
@property
def extracted_summary(self) -> str:
    ...

@property
def extracted_body(self) -> str | None:
    ...
```

But sticking with the original method returning both is reasonable.